src/main/java/frc/robot/subsystems/drive/SwerveModule.java

```diff
@@ -0,0 +1,237 @@
+    SparkPIDController turningPidTmp = turningSparkMax.getPIDController();
+    errors += SparkMaxUtils.check(turningPidTmp.setFeedbackDevice(turningEncoderTmp));
+
+    // Gear ratio 1.0 because the encoder is 1:1 with the module (doesn't involve the actual turning
+    // gear ratio)
+    errors +=
+        SparkMaxUtils.check(
+            SparkMaxUtils.UnitConversions.setRadsFromGearRatio(turningEncoderTmp, DriveConstants.TURN_MODULE_ENCODER_GEAR_RATIO));
+
+    errors +=
+        SparkMaxUtils.check(
+            turningEncoderTmp.setInverted(DriveConstants.TURNING_ENCODER_INVERTED));
+
+    errors += SparkMaxUtils.check(turningPidTmp.setPositionPIDWrappingEnabled(true));
+    errors +=
+        SparkMaxUtils.check(
+            turningPidTmp.setPositionPIDWrappingMinInput(
+                DriveConstants.TURNING_ENCODER_POSITION_PID_MIN_INPUT_RADIANS));
+    errors +=
+        SparkMaxUtils.check(
+            turningPidTmp.setPositionPIDWrappingMaxInput(
+                DriveConstants.TURNING_ENCODER_POSITION_PID_MAX_INPUT_RADIANS));
+
+    errors += SparkMaxUtils.check(turningPidTmp.setP(DriveCal.TURNING_P));
+    errors += SparkMaxUtils.check(turningPidTmp.setI(DriveCal.TURNING_I));
+    errors += SparkMaxUtils.check(turningPidTmp.setD(DriveCal.TURNING_D));
+    errors += SparkMaxUtils.check(turningPidTmp.setFF(DriveCal.TURNING_FF));
+    errors +=
+        SparkMaxUtils.check(
+            turningPidTmp.setOutputRange(
+                DriveCal.TURNING_MIN_OUTPUT, DriveCal.TURNING_MAX_OUTPUT));
+
+    errors +=
+        SparkMaxUtils.check(
+            turningSparkMax.setIdleMode(DriveConstants.TURNING_MOTOR_IDLE_MODE));
+    errors +=
+        SparkMaxUtils.check(
+            turningSparkMax.setSmartCurrentLimit(
+                DriveConstants.TURNING_MOTOR_CURRENT_LIMIT_AMPS));
+
+    return errors == 0;
+  }
+
+  /** Does all the initialization for the spark, return true on success */
+  void initDriveTalon() {
+    TalonFXConfigurator cfg = drivingTalon.getConfigurator();
+    TalonFXConfiguration toApply = new TalonFXConfiguration();
+    toApply.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+    cfg.apply(toApply);
+
+    Slot0Configs slot0Configs = new Slot0Configs();
+    slot0Configs.kP = DriveCal.DRIVING_P; 
+    slot0Configs.kI = DriveCal.DRIVING_I; 
+    slot0Configs.kD = DriveCal.DRIVING_D; 
+    drivingTalon.getConfigurator().apply(slot0Configs); 
+    
+    /*errors +=
+        SparkMaxUtils.check(
+            drivingPidTmp.setOutputRange(
+                DriveCal.DRIVING_MIN_OUTPUT, DriveCal.DRIVING_MAX_OUTPUT));
+
+    errors +=
+        SparkMaxUtils.check(
+            drivingSparkMax.setIdleMode(DriveConstants.DRIVING_MOTOR_IDLE_MODE));
+    errors +=
+        SparkMaxUtils.check(
+            drivingSparkMax.setSmartCurrentLimit(
+                DriveConstants.DRIVING_MOTOR_CURRENT_LIMIT_AMPS));*/ //TODO I cannot find a Talon equivlent for this stuff
```

Comment:
There should be an equivalent to these two. Maybe not the output range one

Reply:
fixed in configurator